docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst

```diff
@@ -0,0 +1,82 @@
+
+Assume the domains a.com and b.com. Settings files can be `base.py`, `acom.py`, and `bcom.py`. The base settings will contain all settings like normal. The contents of site-specific settings override the base settings:
+
+.. code-block:: python
+
+   # acom.py
+   from base import *  # noqa
+
+   ALLOWED_HOSTS = ['a.com']
+   DATABASES["NAME"] = "acom"
+   DATABASES["PASSWORD"] = "password-for-acom"
+   MEDIA_DIR = BASE_DIR / "acom-media"
+
+
+Because each site has its database and media folder, nothing can 'leak' to another site. But this also means that content cannot be shared between sites as one can do when using the multi-site option.
+
+In a multi-instance configuration, each instance requires a certain amount of server resources (CPU and memory). That means adding sites will increase server load. This only scales up to a certain point.
+
+Multi-tenancy
+-------------
+
+Multi-tenancy is a project configuration in which a single instance of the software serves multiple tenants. A tenant is a group of users who have access and permission to a single site. Multitenant software is designed to provide every tenant its configuration, data, and user management.
+
+Wagtail supports *multi-site*, where user management and content are shared. Wagtail can run *multi-instance* where there is full separation of content at the cost of running multiple instances. Multi-tenancy combines the best of both worlds: a single instance, and the full separation of content per site and user management.
+
+Wagtail does not support full multi-tenancy at this moment. But it is on our radar, we would like to improve Wagtail to add multi-tenancy - while still supporting the existing multi-site option. If you have ideas or like to contribute, join us on :ref:`Slack <slack>` in the multi-tenancy channel.
+
+Wagtail currently has the following features to support multi-tenancy:
+
+- A Site model mapping a hostname to a root page
+- Permissions to allow groups of users to manage:
+
+  - arbitrary sections of the page tree
+  - sections of the collection tree (coming soon)
+  - one or more collections of documents and images
+
+- The page API is automatically scoped to the host used for the request
+
+But several features do not currently support multi-tenancy:
+
+- Snippets are global pieces of content so not suitable for multi-tenancy but any model that can be registered as a snippet can also be managed via the Wagtail model admin. You can add a site_id to the model and then use the model admin get_queryset method to determine which site can manage each object. The built-in snippet choosers can be replaced by `modelchooser <https://pypi.org/project/wagtail-modelchooser/>`_ that allows filtering the queryset to restrict which sites may display which objects.
+- Site, site setting, user, and group management. At the moment, your best bet is to only allow superusers to manage these objects.
```

Comment:
Maybe we could add something here as to why this is best left to superusers? Maybe "Permission configuration for built-in models like Sites, Site settings and Users are not site-specific, so any user with permission to edit a single entry can edit them all. This limitation can be mostly circumvented by only allowing superusers to manage these models.

Reply:
Done.